orion-ui/src/compositions/useToast.ts

```diff
@@ -0,0 +1,18 @@
+import { showToast, Toast } from '@prefecthq/prefect-design'
+import { onUnmounted } from 'vue'
+
+/**
+ * The useToast composition returns a callable that can be used in place of showToast to automatically dismiss any toasts created during the component's lifecycle
+ * @returns () => Toast
+ */
+export function useToast(): typeof showToast {
+  const toasts: Toast[] = []
+
+  onUnmounted(() => toasts.forEach(toast => toast.dismiss()))
+
+  return (message, type, options) => {
```

Comment:
can we type these args

```suggestion
  return (...args: Paramaters<typeof showToast>) => {
```

Reply:
Yeah this is more future-proof than just the return type, I like it (didn't commit the suggestion cause the args needed to be passed as well)